### ISSUE TITLE
[lx] /proc/swaps needs to try harder

### ISSUE THREAD

**Issue by rzezeski** [joyent/illumos-joyent#92]
The current emulation of `/proc/swaps` has the correct info but uses the wrong formatting: spaces (lx) vs. tabs (Linux). Consumers that rely on parsing this file may not cope with such a change. E.g. the Oracle 12c Installer segfaults on a NULL pointer because of its inability to parse the different format.

This strace output below is what led me to such a conclusion. I then verified my hypothesis with a modified `lx_proc` module: there is no segfault when `/proc/swaps` is formatted exactly as in Linux [swapfile.c][1].

I'll post a patch either later today or tomorrow.

[1]: http://lxr.free-electrons.com/source/mm/swapfile.c?v=2.6.32#L1704

```
... SNIP ...
35743 open("/tmp/OraInstall2016-01-08_05-28-55PM/command_output_35741", O_WRONLY|O_CREAT|O_TRUNC, 0666) = 5
35743 dup2(5, 1)                        = 1
35743 close(5)                          = 0
35743 dup2(1, 2)                        = 2
35743 fcntl(1, F_GETFD)                 = 0
35743 execve("/bin/cat", ["/bin/cat", "/proc/swaps"], [/* 29 vars */]) = 0
... SNIP ...
35741 stat("/tmp/OraInstall2016-01-08_05-28-55PM/command_output_35741", {st_mode=S_IFREG|0660, st_size=162, ...}) = 0
35741 open("/tmp/OraInstall2016-01-08_05-28-55PM/command_output_35741", O_RDONLY) = 5
35741 fstat(5, {st_mode=S_IFREG|0660, st_size=162, ...}) = 0
35741 mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7ffffe9f0000
35741 read(5, "Filename                        "..., 512) = 162
35741 close(5)                          = 0
35741 munmap(0x7ffffe9f0000, 4096)      = 0
35741 --- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0} ---
35741 +++ killed by SIGSEGV (core dumped) +++
```

**Comment by pfmooney** [comment]
I'll take a look whenever you post patch.

**Comment by pfmooney** [comment]
Open as [OS-5083](https://smartos.org/bugview/OS-5083)

**Comment by rzezeski** [comment]
Webrev: http://zinascii.com/pub/illumos/joyent/5083/

Test program:

```C
#include <stdio.h>
#include <strings.h>

int
main(int argc, char **argv)
{
        long long unsigned totswap = 16 * 1024 * 1024;
        long long unsigned usedswap = 16 * 1024 * 1024;

        printf("Filename\t\t\t\tType\t\tSize\tUsed\tPriority\n");
        printf("%-40s%s\t%llu\t%llu\t%d\n",
            "/illumos/dev/swap", "partition", totswap, usedswap, -1);

        char linux[256] = "/linux/dev/swap";
        size_t len = strlen(linux);

        printf("%s", linux);
        printf("%*s%s\t%u\t%u\t%d\n",
            len < 40 ? 40 - len : 1, " ",
            "partition",
            (unsigned)totswap,
            (unsigned)usedswap,
            -1);

        return (0);
}
```

Test output:

```
[root@sos-live-wrk ~]# ./test-swaps
Filename                                Type            Size    Used    Priority
/illumos/dev/swap                       partition       16777216        16777216        -1
/linux/dev/swap                         partition       16777216        16777216        -1
```

**Comment by pfmooney** [comment]
Merged in 931583b7c29efe8ef215cb94fe4173f1848fca6f

Thanks
